src/core/message-parser.ts

```diff
@@ -41,17 +62,71 @@ export function parseMessage(roomId: string | undefined, message: Ably.InboundMe
     throw new Ably.ErrorInfo(`received incoming message without extras`, 50000, 500);
   }
 
-  if (!messageCreatedMessage.extras.timeserial) {
+  if (!messageCreatedMessage.serial) {
```

Comment:
Can we rename `messageCreatedMessage` variable since it's no longer just new messages that get here?